scripts/metavar.py

```diff
@@ -363,14 +370,17 @@ class Var(object):
                     VariableProperty('dimensions', list,
                                      check_fn_in=check_dimensions),
                     VariableProperty('type', str,
-                                     check_fn_in=check_fortran_type),
+                                     # DH* Avoid checking of DDTs being registered or not,
+                                     # this only works if all DDTs required in one metadata
+                                     # are defined in that file and/or parsed beforehand.
+                                     #check_fn_in=check_fortran_type),
+                                     # *DH
+                                     ),
```

Comment:
We discussed this either in the original PR that introduced the change or in the ccpp-framework meeting. The current version of the parsing scripts for the new metadata rely on all DDTs referenced in one metadata table to be declared in the same metadata table (and also the order matters, they have to be declared beforehand). This doesn't work when radiation sub-DDTs are declared elsewhere and referenced in GFS_typedefs.meta. There won't be a simple fix for this, because we are only reading one (new) metadata file at a time with `ccpp_prebuild.py`. 

I am also wondering if making the following change in check_fortran_type makes any difference in terms of how big the change is:

Turn
```
    dt = ""
    match = check_fortran_intrinsic(typestr, error=False)
    if match is None:
        match = registered_fortran_ddt_name(typestr)
        dt = " derived"
    # End if
    if match is None:
        if error:
            raise CCPPError("'{}' is not a valid{} Fortran type".format(typestr, dt))
        else:
            typestr = None
        # End if
    # End if
    return typestr
```
to
```
    dt = ""
    match = check_fortran_intrinsic(typestr, error=False)
    if match is None:
        match = registered_fortran_ddt_name(typestr)
        dt = " derived"
    # End if
    #if match is None:
    #    if error:
    #        raise CCPPError("'{}' is not a valid{} Fortran type".format(typestr, dt))
    #    else:
    #        typestr = None
    #    # End if
    # End if
    return typestr
```
?

Reply:
That is what I was thinking, at least for now. We should work towards a more robust solution that takes various host build systems into account.